File: database-plaything.py
"""Simple little program that takes name and age and puts them in a
   database, or views DB in nicely formatted manner. Why, you ask?
   Because learning, that's why."""

import sqlite3
import os.path

DATABASE_NAME = "people-to-kill.db"
PROMPT = ">>> "
CONNECTION = None
CURSOR = None

def valid_age(age):
    """checks if age (string) is int and above > 0"""
    if age.isdigit():
        return True
    else:
        return False 

def valid_name(name):
    """checks to make sure name < 64 characters, because anyone
       who has a longer name than that cannot be trusted."""
    if len(name) > 0 and len(name) < 64:
        return True
    else:
        return False

def database_exists(dbname):
    """Checks if database exists in current directory"""
    print("Creating database " + dbname)
    return os.path.isfile(dbname)

def create_database(dbname):
    """create database"""
    global CONNECTION
    global CURSOR
    CONNECTION = sqlite3.connect(DATABASE_NAME)
    CURSOR = CONNECTION.cursor()
    CURSOR.execute("""CREATE TABLE people
                 (name text, age integer)""")
    CONNECTION.commit()

def insert_data(name, age):
    """opens database, puts shit into it, commits and closes"""
    CURSOR.execute("""INSERT INTO people VALUES
                 (?, ?)""", (name, int(age)))
    CONNECTION.commit()

def view_data():
    """Prints the database contents."""
    table = []
    for row in CURSOR.execute("SELECT * FROM people ORDER BY name"):
        table.append(row)
    table.sort()
    print("Contents of " + DATABASE_NAME + ":")
    print("NAME | AGE")
    for entry in table:
        print("{} | {}".format(entry[0], entry[1]))


def take_input():
    """Gets name and age."""
    user_input = ""
    while user_input != "exit" and not valid_name(user_input):
        print("Name, please:")
        name = user_input = input(PROMPT)
    while user_input != "exit" and not valid_age(user_input):
        print("Age, please:")
        age = user_input = input(PROMPT)
    if user_input != "exit":
        insert_data(name, age)
        print("'{},' age {}, inserted into database.".format(name, age)) 

def main():
    """Sets up some shit and does the main loop."""
    if not database_exists(DATABASE_NAME):
        create_database(DATABASE_NAME)
    else:
        global CONNECTION
        global CURSOR
        CONNECTION = sqlite3.connect(DATABASE_NAME)
        CURSOR = CONNECTION.cursor()
    print("""Hello! Type "view" to view your database, or "insert" """)
    print("""to add a name and age. Type "exit" at any point to quit.""")
    user_input = ""
    while user_input != "exit":
        user_input = input(PROMPT).lower()
        if user_input == "insert":
            take_input()
        elif user_input == "view":
            view_data()
        else:
            if user_input != "exit":
                print("ERROR CODE ID-10-t; please contact system administrator.")
    print("Have a nice day!")

if __name__ == "__main__":
    main() 

File: shitty-java-programs/databasePlaything.java
import java.sql.*;
import java.util.ArrayList;

public class databasePlaything 
{
    //so java has no built in sqlite library, and I'm
    //too lazy to fuck around with that shit, so here's
    //a virtual database.
    public static ArrayList<String[]> database = new ArrayList<String[]>();

    public static boolean validAge(String age) 
    {
        try // this try/catch block returns false if it's not
        {   // convertible to an int
            if (Integer.parseInt(age) > 0)
            {
                return true;
            }
            return false;

        } 
        catch(NumberFormatException e) 
        { 
            return false; 
        }
    }

    public static boolean validName(String name) 
    {
        if (name.length() > 0 && name.length() < 64) 
        {
            return true;
        }
        else {
            return false;
        }
    }

    public static void insertData(String name, String age) 
    {
        String[] input = {name, age};
        database.add(input);

    }

    public static void viewData() 
    {
        for (int i = 0; i < database.size(); i++) {
            System.out.println(database.get(i)[0] + " " + database.get(i)[1]);
        }
    }

    public static void takeInput() 
    {
        String userInput = "";
        String name = "";
        String age = "";
        while (userInput != "exit" && !validName(name)) 
        {
            userInput = System.console().readLine("Name, please: ");
            name = userInput;
        }
        while (userInput != "exit" && !validAge(age)) 
        {
            userInput = System.console().readLine("Age, please: ");
            age = userInput;
        }
        insertData(name, age);
    }
    
    public static void main(String[] args) 
    {
        System.out.println("Welcome to Critter's amazing database thing!");
        System.out.println("'insert' to add entries, 'view' to view shit");
        String userInput = "";
        while (userInput != "exit") 
        {
            userInput = System.console().readLine(">>> ");
            if (userInput.equals("insert"))
            {
                takeInput();
            }
            else if (userInput.equals("view"))
            {
                viewData();    
            }

        }
    }

}